### PR TITLE
Add docker-compose.prod.yml and .env.template, fix prod hostname port

### PR DESCRIPTION
**Pull request by airelawaleria**

### CODE REVIEW

File: docker-compose.prod.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - ./letsencrypt:/letsencrypt
+
+  server:
+    image: "ghcr.io/ls1intum/prompt/prompt-server:latest"
+    container_name: orgatool-server
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.server.rule=Host(`prompt.ase.cit.tum.de`) && PathPrefix(`/api`)"
+      - "traefik.http.routers.server.entrypoints=websecure"
+      - "traefik.http.routers.server.tls.certresolver=letsencrypt"
```
Comment (robertjndw) [inline docker-compose.prod.yml:31]: You have to add one more label to tell Traefik, that the service is running on a different ports (in your case 8080)
`- "traefik.http.services.server.loadbalancer.server.port=8080"`

File: docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      - "--entrypoints.web.http.redirections.entrypoint.scheme=https"
+      - "--certificatesresolvers.letsencrypt.acme.httpchallenge=true"
+      - "--certificatesresolvers.letsencrypt.acme.httpchallenge.entrypoint=web"
+      - "--certificatesresolvers.letsencrypt.acme.email=valeryia.andraichuk@tum.de"
+      - "--certificatesresolvers.letsencrypt.acme.storage=/letsencrypt/acme.json"
+    restart: unless-stopped
+    ports:
+      - "80:80"
+      - "443:443"
+      - "8080:8080"
```
Comment (robertjndw) [inline docker-compose.prod.yml:19]: I don't think that is necessary anymore. 80 and 443 should be enough as your server is running on `/api` path now

File: docker-compose.prod.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - ./letsencrypt:/letsencrypt
+
+  server:
+    image: "ghcr.io/ls1intum/prompt/prompt-server:latest"
+    container_name: orgatool-server
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.server.rule=Host(`prompt.ase.cit.tum.de`) && PathPrefix(`/api`)"
```
Comment (robertjndw) [inline docker-compose.prod.yml:29]: When you add the prefix, you maybe need to cut the prefix from the requests
```
- "traefik.http.middlewares.strip-prefix.stripprefix.prefixes=/api"
- "traefik.http.routers.server.middlewares=strip-prefix"
```

**Review by robertjndw — CHANGES_REQUESTED** [review]

File: docker-compose.prod.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      - "8080"
+    environment:
+      - SPRING_DATASOURCE_URL=${SPRING_DATASOURCE_URL}
+      - SPRING_DATASOURCE_USERNAME=${SPRING_DATASOURCE_USERNAME}
+      - SPRING_DATASOURCE_PASSWORD=${SPRING_DATASOURCE_PASSWORD}
+      - SPRING_JPA_HIBERNATE_DDL_AUTO=update
+      - CLIENT_HOST=${CLIENT_HOST}
+      - KEYCLOAK_CLIENT_SECRET=${KEYCLOAK_CLIENT_SECRET}
+      - KEYCLOAK_ISSUER_URI=${KEYCLOAK_ISSUER_URI}
+      - KEYCLOAK_JWK_SET_URI=${KEYCLOAK_JWK_SET_URI}
```
Comment (robertjndw) [inline docker-compose.prod.yml:45]: This is unneccessary. Docker reads the variables automatically from the `.env` file.
See https://docs.docker.com/compose/environment-variables/set-environment-variables/

Comment (robertjndw) [reply docker-compose.prod.yml:45]: Should work like this too
```suggestion
      - SPRING_DATASOURCE_URL
      - SPRING_DATASOURCE_USERNAME
      - SPRING_DATASOURCE_PASSWORD
      - SPRING_JPA_HIBERNATE_DDL_AUTO=update
      - CLIENT_HOST
      - KEYCLOAK_CLIENT_SECRET
      - KEYCLOAK_ISSUER_URI
      - KEYCLOAK_JWK_SET_URI
```

File: docker-compose.prod.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.client.rule=Host(`prompt.ase.cit.tum.de`)"
+      - "traefik.http.routers.client.entrypoints=websecure"
+      - "traefik.http.routers.client.tls.certresolver=letsencrypt"
+    expose:
+      - "80"
+    environment:
+      - SERVER_HOST=${SERVER_HOST}
+      - KEYCLOAK_HOST=${KEYCLOAK_HOST}
```
Comment (robertjndw) [inline docker-compose.prod.yml:75]: Same here

File: .env.template [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+SPRING_DATASOURCE_USERNAME=
+SPRING_DATASOURCE_PASSWORD=
+
+SERVER_HOST=
+CLIENT_HOST=
+
+KEYCLOAK_HOST=
+KEYCLOAK_ISSUER_URI=
+KEYCLOAK_JWK_SET_URI=
+KEYCLOAK_CLIENT_SECRET=
```
Comment (robertjndw) [inline .env.template:18]: I would personally enter some sample values here, particularly for the URLs.

File: docker-compose.prod.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
+services:
```
Comment (robertjndw) [inline docker-compose.prod.yml:1]: Please put a version here: `version: '3.8'`

File: docker-compose.prod.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      - POSTGRES_PASSWORD=${SPRING_DATASOURCE_PASSWORD}
+    healthcheck:
+      test:
+        [
+          "CMD-SHELL",
+          "pg_isready -d $$SPRING_DATASOURCE_USERNAME -U $$SPRING_DATASOURCE_PASSWORD",
+        ]
+      interval: 5s
+      timeout: 5s
+      retries: 5
```
Comment (robertjndw) [inline docker-compose.prod.yml:61]: Your database has no volumes assigned to it. If you restart the container, all your data will be lost.

File: docker-compose.prod.yml (hunk 1 above)
Comment (airelawaleria) [reply docker-compose.prod.yml:33]: done, however, it works also without this line

File: docker-compose.prod.yml (hunk 2 above)
Comment (airelawaleria) [reply docker-compose.prod.yml]: done

File: docker-compose.prod.yml (hunk 3 above)
Comment (airelawaleria) [reply docker-compose.prod.yml:30]: i tried it and it breaks the config

File: docker-compose.prod.yml (hunk 4 above)
Comment (airelawaleria) [reply docker-compose.prod.yml:47]: didn't work, got errors that env variables are not passed

File: .env.template (hunk 6 above)
Comment (airelawaleria) [reply .env.template]: done

File: docker-compose.prod.yml (hunk 7 above)
Comment (airelawaleria) [reply docker-compose.prod.yml:3]: done

File: docker-compose.prod.yml (hunk 8 above)
Comment (airelawaleria) [reply docker-compose.prod.yml]: done

File: docker-compose.prod.yml (hunk 4 above)
Comment (robertjndw) [reply docker-compose.prod.yml:47]: Okay, then you can leave it as is. It was just a styling suggestion.

**Review by robertjndw — APPROVED** [review]

**Pull request merged**